compiler/rustc_target/src/abi/mod.rs

```diff
@@ -1121,38 +1121,44 @@ pub enum Variants<'a> {
 
     /// Enum-likes with more than one inhabited variant: each variant comes with
     /// a *discriminant* (usually the same as the variant index but the user can
-    /// assign explicit discriminant values).  That discriminant is encoded
-    /// as a *tag* on the machine.  The layout of each variant is
-    /// a struct, and they all have space reserved for the tag.
-    /// For enums, the tag is the sole field of the layout.
+    /// assign explicit discriminant values). That discriminant is encoded
+    /// in a field of the object. The layout of each variant is
+    /// a struct. For enums, the tag is the sole field of the layout.
     Multiple {
-        tag: Scalar,
-        tag_encoding: TagEncoding,
-        tag_field: usize,
+        encoding: TagEncoding,
+        scalar: Scalar,
+        field: usize,
```

Comment:
I could live with removing the `tag_` prefix (but it seems you did a global search-and-replace here and that changed the meaning of some code, see my previous comment), but `scalar: Scalar` is worse than before IMO. This is the scalar ABI of the tag.

You also removed the term 'tag' from the comment, which is bad IMO. We used to have a lot of confusion between discriminant and tag, until I started systematically putting these terms everywhere. Please don't revert us back to the confusing state.